### PR TITLE
Add AWS resource tag with the Giant Swarm Installation

### PR DESCRIPTION
**Pull request by rossf7**
Towards giantswarm/giantswarm#2775

Adds the installation tag which we want to use for cost tracking. This is configured via a flag so it needed to be added to the `kmskey` and `s3bucket` resources. 

Maybe we should add the installation name to the provider CRDs as IMO it's useful metadata. I've added to the cluster object refactoring issue.

https://github.com/giantswarm/giantswarm/issues/2383#issuecomment-371749665

### CODE REVIEW

File: service/awsconfig/v8/cluster_resource_set.go [hunk 1]
```diff
@@ -147,6 +147,8 @@ func NewClusterResourceSet(config ClusterResourceSetConfig) (*framework.Resource
 				KMS: config.GuestAWSClients.KMS,
 			},
 			Logger: config.Logger,
+
+			InstallationName: config.InstallationName,
```
Comment (rossf7) [inline service/awsconfig/v8/cluster_resource_set.go]: We need to pass the installation name to the `kmskey` and `s3bucket` resources.

File: service/awsconfig/v8/key/key.go [hunk 2]
```diff
@@ -142,11 +145,12 @@ func ClusterOrganization(customObject v1alpha1.AWSConfig) string {
 	return ClusterCustomer(customObject)
 }
 
-func ClusterTags(customObject v1alpha1.AWSConfig) map[string]string {
+func ClusterTags(customObject v1alpha1.AWSConfig, installationName string) map[string]string {
 	cloudProviderTag := ClusterCloudProviderTag(customObject)
 	tags := map[string]string{
 		cloudProviderTag:    CloudProviderTagOwnedValue,
 		ClusterTagName:      ClusterID(customObject),
+		InstallationTagName: installationName,
```
Comment (rossf7) [inline service/awsconfig/v8/key/key.go]: Here the new tag is added.

File: service/awsconfig/v8/resource/cloudformation/resource.go [hunk 3]
```diff
@@ -70,8 +70,8 @@ func (r *Resource) Name() string {
 	return Name
 }
 
-func getCloudFormationTags(customObject v1alpha1.AWSConfig) []*awscloudformation.Tag {
-	clusterTags := key.ClusterTags(customObject)
+func (r *Resource) getCloudFormationTags(customObject v1alpha1.AWSConfig) []*awscloudformation.Tag {
```
Comment (rossf7) [inline service/awsconfig/v8/resource/cloudformation/resource.go]: Changed to a method to use the configured installation name.

File: service/awsconfig/v8/version_bundle.go [hunk 4]
```diff
@@ -14,6 +14,11 @@ func VersionBundle() versionbundle.Bundle {
 				Description: "Fix validation check for existing peering routes.",
 				Kind:        versionbundle.KindFixed,
 			},
+			{
+				Component:   "aws-operator",
+				Description: "Add AWS resource tag with the Giant Swarm Installation.",
+				Kind:        versionbundle.KindChanged,
+			},
```
Comment (rossf7) [inline service/awsconfig/v8/version_bundle.go]: Changelog.

File: service/awsconfig/v8/key/key_test.go [hunk 5]
```diff
@@ -156,10 +156,13 @@ func Test_ClusterOrganization(t *testing.T) {
 }
 
 func Test_ClusterTags(t *testing.T) {
+	installName := "test-install"
+
 	expectedID := "test-cluster"
 	expectedTags := map[string]string{
 		"kubernetes.io/cluster/test-cluster": "owned",
 		"giantswarm.io/cluster":              "test-cluster",
+		"giantswarm.io/installation":         "test-install",
```
Comment (kopiczko) [inline service/awsconfig/v8/key/key_test.go]: WDYT about naming it `giantswarm.io/host-cluster` to be more Giant Swarm agnostic and more usable outside. Also we use the term in several places already.

**Comment by rossf7** [comment]
@pipo02mix Yes I think adding it to the CRD makes sense. I've added it to #2383 which tracks changes we want to make to the cluster object.

File: service/awsconfig/v8/key/key_test.go (hunk 5 above)
Comment (rossf7) [reply service/awsconfig/v8/key/key_test.go]: I'm OK with it as long as we use the same tag in guest and host clusters.

Comment (kopiczko) [reply service/awsconfig/v8/key/key_test.go]: I don't see it being a problem. Do you?

File: service/awsconfig/v8/resource/cloudformation/resource_test.go [hunk 6]
```diff
@@ -33,13 +42,45 @@ func Test_Resource_Cloudformation_GetCloudFormationTags(t *testing.T) {
...
+					Value: aws.String("test-install"),
+				},
 			},
 		},
 	}
 
 	for _, tc := range testCases {
 		t.Run(tc.description, func(t *testing.T) {
-			tags := getCloudFormationTags(tc.obj)
+			c := Config{}
```
Comment (fgimenez) [inline service/awsconfig/v8/resource/cloudformation/resource_test.go]: Maybe all the config setup until the installation name could be extracted out of the loop, it doesn't depend on `tc`, WDYT?

File: service/awsconfig/v8/key/key_test.go (hunk 5 above)
Comment (rossf7) [reply service/awsconfig/v8/key/key_test.go]: No should be fine let's change it.

Comment (xh3b4sd) [reply service/awsconfig/v8/key/key_test.go]: Our wording in opsctl is installation. Not sure what the reason for @marians was to do so.

Comment (rossf7) [reply service/awsconfig/v8/key/key_test.go]: @kopiczko I'm staying with installations as requested by Timo which aligns with the installations repo and opsctl.

See https://github.com/giantswarm/giantswarm/issues/2775#issuecomment-371757090

Comment (kopiczko) [reply service/awsconfig/v8/key/key_test.go]: opsctl is 100% internal tool. It's fine to use whatever term there.

File: service/awsconfig/v8/resource/cloudformation/resource_test.go (hunk 6 above)
Comment (rossf7) [reply service/awsconfig/v8/resource/cloudformation/resource_test.go]: Good point! Done.

File: service/awsconfig/v8/key/key_test.go (hunk 5 above)
Comment (rossf7) [reply service/awsconfig/v8/key/key_test.go]: Let's discuss in #2775.

**Pull request merged**